### ISSUE TITLE
write_graphite plugin: DS type does not match value list type

### ISSUE THREAD

**Issue by esala116** [collectd/collectd#1996]
*   Version of collectd:
*   Operating system / distribution:

## Expected behavior

collectd send data to graphite

(Description of the behavior / output that you expected)

## Actual behavior

write_graphite plugin started failure

(Description of the behavior / output that you observed)

## Steps to reproduce

*   step 1  version: collectd-5.6.1
*   step 2 "write_graphite plugin: DS type does not match value list type" in syslog
*   step 3 conf as follow:
<Plugin write_graphite>
  <Carbon>
    Host "10.180.158.233"
    Port 2003
    Prefix "collectd."
    EscapeCharacter "_"
    StoreRates true
    AlwaysAppendDS false
    SeparateInstances false
    LogSendErrors true
  </Carbon>
</Plugin>

**Comment by psschroeter** [comment]
This issue affects write_http as well for collectd 5.6.1. So I was looking at this commit: https://github.com/collectd/collectd/commit/2e80174b4aa0b983c193453b1062677a0d28188c#diff-6cc56e464de9455059087c05f3b780f8

And I'm very confused. The code went from:
`if (0 != strcmp (ds->type, vl->type)) { ERROR("DS type does not match...") }`

To:
`if (strcmp (ds->type, vl->type) == 0) { ERROR ("DS type does not match...") }`

The logic was reversed.... can this be right? Nothing seems to work for collectd 5.6 no matter I do!

**Comment by rpv-tomsk** [comment]
@rubenk ?

**Comment by rubenk** [comment]
@rpv-tomsk Yeah, we have several ;) No need to ping us on every issue.

In any case, I think @psschroeter is right and this was a simple oversight.
I'm a bit short on time these days, can any of you do a PR against the 5.6 branch?

**Comment by rpv-tomsk** [comment]
>No need to ping us on every issue.

Ok.

PR created.

**Comment by tzhang2017** [comment]
Hi,
I have the save message. For information I have installed uncer Centos 7 via yum collectd 5.7 and rrdtool 1.4.8. My configuration file is
Hostname    "localhost"
BaseDir     "/var/lib/collectd"
PIDFile     "/var/run/collectd.pid"
TypesDB     "/usr/share/collectd/types.db"
Interval        10
LoadPlugin logfile
<Plugin logfile>
        LogLevel info
        File "/var/log/collectd.log"
        Timestamp true
        PrintSeverity false
</Plugin>
LoadPlugin cpu
LoadPlugin memory
LoadPlugin rrdtool
<Plugin rrdtool>
        DataDir "/var/lib/collectd/rrd"
        CreateFilesAsync false
        CacheTimeout 120
        CacheFlush   900
        WritesPerSecond 50
</Plugin>

And I got logs:

[2017-05-15 16:18:44] collectd: Stopping 5 read threads.
[2017-05-15 16:18:44] collectd: Stopping 5 write threads.
[2017-05-15 16:18:44] rrdtool plugin: Shutting down the queue thread.
[2017-05-15 16:18:44] plugin_load: plugin "logfile" successfully loaded.
[2017-05-15 16:18:44] plugin_load: plugin "cpu" successfully loaded.
[2017-05-15 16:18:44] plugin_load: plugin "memory" successfully loaded.
[2017-05-15 16:18:44] plugin_load: plugin "rrdtool" successfully loaded.
[2017-05-15 16:18:44] Systemd detected, trying to signal readyness.
[2017-05-15 16:18:44] Initialization complete, entering read-loop.
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:44] Filter subsystem: Built-in target `write': Dispatching value to all write plugins failed with status -1.
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:44] rrdtool plugin: DS type does not match value list type
[2017-05-15 16:18:54] rrdtool plugin: DS type does not match value list type


And I have no rrd files generated!!

Other tests have also been done:
- the rrdtool is working fine manually
- the collectd works fine with csv ouput plugin 


I would like to know if you have resolved this issue. I've been blocked for serveral days.....
Thanks a lot!!
